resources/utils/user.go

```diff
@@ -65,7 +65,18 @@ type Group struct {
 // lineFunc returns a value, an error, or (nil, nil) to skip the row.
 type lineFunc func(line []byte) (v interface{}, err error)
 
-func GetUserNameFromID(uid uint32, userFilePath string) (string, error) {
+type OSUser interface {
+	GetUserNameFromID(uid uint32, userFilePath string) (string, error)
+	GetGroupNameFromID(gid uint32, groupFilePath string) (string, error)
+}
+
+type OSUserUtil struct{}
```

Comment:
Wondering about this struct.

1. The function is a `pure` function - for the same input, it will always return the same output.
2. The function has no side effects.
3. The struct is stateless.
4. We do not modify any variable after assigning a value.

It leads me to think, that a struct is not necessary here and we can use functional programming in this case.
It also will reduce the need to mock this class every time we test file-system and will use the real functionality instead.

WDT ?

Reply:
After a discussion we had, I decided to keep the current implementation.
Mocking the util in the unit tests is worth the overhead of creating an interface for a utility.
Unit tests suppose to test the class functionality, as the utility has unit tests of its own, it should be mocked in this use case.